Source/portal.cpp

```diff
@@ -113,16 +113,15 @@ bool PortalOnLevel(int i)
 
 void RemovePortalMissile(int id)
 {
-	for (int i = 0; i < ActiveMissileCount; i++) {
-		int mi = ActiveMissiles[i];
-		auto &missile = Missiles[mi];
+	for (auto &missileWithId : Missiles) {
+		auto &missile = missileWithId.second;
 		if (missile._mitype == MIS_TOWN && missile._misource == id) {
 			dFlags[missile.position.tile.x][missile.position.tile.y] &= ~DungeonFlag::Missile;
 
 			if (Portals[id].level != 0)
 				AddUnLight(missile._mlid);
 
-			DeleteMissile(i);
+			DeleteMissile(missileWithId.first);

```

Comment:
This is the only use of `DeleteMissile`. This should use `Missiles.erase(iterator)` directly since it's looping over the collection.

Reply:
Also just realised after looking at the above, this is another case where `std::remove_if` could be used. The current code doesn't return early so it matches the current logic and means you don't need to manage the iterator manually.

Reply:
I inlined `Missiles.erase` and made a `break`/early return if we found a town portal for the user.
Cause there is only one possible town portal per player.